allensdk/internal/api/behavior_lims_api.py

```diff
@@ -44,21 +59,24 @@ def get_extended_trials(self):
     @staticmethod
     def foraging_id_to_behavior_session_id(foraging_id):
         '''maps foraging_id to behavior_session_id'''
-        api = PostgresQueryMixin()
+        api = (credential_injector(LIMS_DB_CREDENTIAL_MAP)
+               (PostgresQueryMixin)())
         query = '''select id from behavior_sessions where foraging_id = '{}';'''.format(
             foraging_id)
         return api.fetchone(query, strict=True)
 
     @staticmethod
     def behavior_session_id_to_foraging_id(behavior_session_id):
         '''maps behavior_session_id to foraging_id'''
-        api = PostgresQueryMixin()
+        api = (credential_injector(LIMS_DB_CREDENTIAL_MAP)
+               (PostgresQueryMixin)())
         query = '''select foraging_id from behavior_sessions where id = '{}';'''.format(
             behavior_session_id)
         return api.fetchone(query, strict=True)
 
     @classmethod
-    def from_foraging_id(cls, foraging_id):
+    def from_foraging_id(cls, foraging_id: str,
+                         lims_credentials: Optional[DbCredentials] = None):
```

Comment:
lims_credentials should be passed to the class constructor?

Reply:
Do you think it shouldn't be? We do need to initialize the LIMS connection to create the class. I wanted to expose the credentials to the class constructor so that users could optionally provide the credentials, rather than have the environment variable provider be the only option.

Reply:
I put this comment on one line too high, I meant that it looks like you forgot to add lims_credentials to the line below

Reply:
Nice catch!